src/utils.py

```diff
@@ -11,6 +12,21 @@ def create_directory(path):
     else:
         print(f"Local directory '{path}' already exsists.")
 
+def ensure_parent_containers(tiled_uri, tiled_api_key):
+    parsed_url = urlparse(tiled_uri)
+    path = parsed_url.path
+    # Splitting path into parts
+    path_parts = path.split('/')[1:]  # Split and remove the first empty element
+    tiled_root = f'{parsed_url.scheme}://{parsed_url.netloc}/{path_parts[0]}/{path_parts[1]}/{path_parts[2]}'
```

Comment:
it is much safer to let python's urllib class create the url:

tiled_root = urlunparse(
    (parsed_url.scheme, parsed_url.netloc, new_path)
)

Reply:
Addressed in the latest push